src/EluvioLive.js

```diff
@@ -1347,9 +1348,63 @@ class EluvioLive {
       }
 
     } else if (showOwners && showOwners > 0) {
-      var maxShowOwners = showOwners;
+      // prefer this fast one to the contract code, which is left for reference below
+      const maxShowOwners = showOwners;
+      nftInfo.tokens = [];
+
+      try {
+        const url = urljoin("/nft/owners/", addr);
+        let res = await this.GetServiceRequest({
+          path: url,
+          queryParams: { limit: maxShowOwners },
+        });
+        res = await res.json();
+
+        nftInfo.tokens = res.contents;
+      } catch (e) {
+        warns.push("Failed to get token owners: " + addr + " - " + e);
+      }
 
       for (var i = 0; i < maxShowOwners && i < nftInfo.totalSupply; i++) {
+        try {
+          // adapt to format:
+          // hold -> holdSecs
+          nftInfo.tokens[i].holdSecs = nftInfo.tokens[i].hold;
+          // add holdEnd
+          const timestamp = nftInfo.tokens[i].hold;
+          nftInfo.tokens[i].holdEnd = new Date(timestamp * 1000);
+          // token_id -> tokenId
+          nftInfo.tokens[i].tokenId = "" + nftInfo.tokens[i].token_id;
+          // token_uri -> tokenURI
+          nftInfo.tokens[i].tokenURI = nftInfo.tokens[i].token_uri;
+          // token_owner -> owner
+          nftInfo.tokens[i].owner = nftInfo.tokens[i].token_owner;
+          // ordinal int -> string
+          nftInfo.tokens[i].ordinal = "" + nftInfo.tokens[i].ordinal;
+
+          delete nftInfo.tokens[i].block;
```

Comment:
instead of setting everything and then deleting, can set one at a time from res.contents.  will change